larky/src/main/resources/stdlib/xml/etree/ElementTree.star

```diff
@@ -941,72 +955,68 @@ def _namespaces(elem, default_namespace=None):
                 add_qname(tag)
         elif tag != None and tag != Comment and tag != PI:
             _raise_serialization_error(tag)
-        for key, value in elem.items():
+        for key, value in current.items():
             if types.is_instance(key, QName):
                 key = key.text
             if key not in qnames:
                 add_qname(key)
             if types.is_instance(value, QName) and value.text not in qnames:
                 add_qname(value.text)
-        text = elem.text
+        text = current.text
         if types.is_instance(text, QName) and text.text not in qnames:
             add_qname(text.text)
+        qu.extend(current._children)
     return qnames, namespaces
 
+def prepare_elems_for_serialize(tops_level_elems):
```

Comment:
Wouldn't this function be called `flatten_nested_elements` and introduce a docstring to say that we're building an iterative traversal due to lack of recursion support?

Reply:
updated